### PR TITLE
Patch avahi for multiple CVEs

### PR DESCRIPTION
**Pull request by Kanishk-Bansal**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- CVE-2023-38469
- CVE-2023-38470
- CVE-2023-38471
- CVE-2023-38472
- CVE-2023-38473

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-38469
- https://nvd.nist.gov/vuln/detail/CVE-2023-38470
- https://nvd.nist.gov/vuln/detail/CVE-2023-38471
- https://nvd.nist.gov/vuln/detail/CVE-2023-38472
- https://nvd.nist.gov/vuln/detail/CVE-2023-38473

###### Test Methodology

- Pipeline build id: [686618](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=686618&view=results)

### CODE REVIEW

File: SPECS/avahi/avahi.spec [hunk 1]
```diff
@@ -405,6 +410,21 @@ exit 0
 %endif
 
 %changelog
+* Mon Dec 02 2024 Kanishk Bansal <kanbansal@microsoft.com> -0.8-2
```
Comment (anphel31) [inline SPECS/avahi/avahi.spec:413]: Can you please combine all of these comments into a single changelog entry? That is how we typically mention multiple patches with the same release number.

Comment (Kanishk-Bansal) [reply SPECS/avahi/avahi.spec:413]: done

**Review by rikenm1 — APPROVED** [review]

File: SPECS/avahi/avahi.spec [hunk 2]
```diff
@@ -405,6 +410,13 @@ exit 0
...
```
[the remark this answers is not in the document]
Comment (Kanishk-Bansal) [reply SPECS/avahi/avahi.spec]: done

**Comment by rikenm1** [comment]
@Kanishk-Bansal, can you fix the merge conflict for this ?

File: SPECS/avahi/avahi.spec [hunk 3]
```diff
@@ -3,7 +3,7 @@
 Summary:        Local network service discovery
 Name:           avahi
 Version:        0.8
-Release:        2%{?dist}
+Release:        3{?dist}
```
Comment (rikenm1) [inline SPECS/avahi/avahi.spec:6]:
```suggestion
Release:        3%{?dist}
```

**Review by rikenm1 — CHANGES_REQUESTED** [review]

**Review by jslobodzian — APPROVED** [review]

**Comment by nicogbg** [comment]
Auto cherry-pick results:
- main :white_check_mark: -> https://github.com/microsoft/azurelinux/pull/11606

Auto cherry-pick pipeline run -> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=697912&view=results

**Pull request merged**